crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -1521,6 +1521,32 @@ impl Segment2d {
         let t = projection_scaled / length_squared;
         self.vertices[0] + t * segment_vector
     }
+
+    /// Returns the point of intersection between this [`Segment2d`] and another, if it exists.
+    #[inline(always)]
+    pub fn intersect(&self, other: Self) -> Option<Vec2> {
+        let p = self.point1();
+        let q = other.point1();
+        let r = self.scaled_direction();
+        let s = other.scaled_direction();
+
+        let r_cross_s = r.perp_dot(s);
+        let q_minus_p = q - p;
+
+        if r_cross_s != 0.0 {
+            let t = q_minus_p.perp_dot(s / r_cross_s);
+            let u = q_minus_p.perp_dot(r / r_cross_s);
```

Comment:
```suggestion
            let t = q_minus_p.perp_dot(s) / r_cross_s;
            let u = q_minus_p.perp_dot(r) / r_cross_s;
```
This saves a bunch of floating point operations by dividing resulting numbers instead of the vectors.